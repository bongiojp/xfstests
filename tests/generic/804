#! /bin/bash
# SPDX-License-Identifier: GPL-2.0
# Copyright 2019 Google LLC
#
# FS QA Test generic/804
#
# Verify ciphertext for v2 encryption policies that use Adiantum to encrypt file
# contents and file names.
#
# This is the same as generic/550, except using v2 policies.
#
seq=`basename $0`
seqres=$RESULT_DIR/$seq
echo "QA output created by $seq"

here=`pwd`
tmp=/tmp/$$
status=1	# failure is the default!
trap "_cleanup; exit \$status" 0 1 2 3 15

_cleanup()
{
	cd /
	rm -f $tmp.*
}

# get standard environment, filters and checks
. ./common/rc
. ./common/filter
. ./common/encrypt

# remove previous $seqres.full before test
rm -f $seqres.full

# real QA test starts here
_supported_fs generic
_supported_os Linux

# Test both with and without the DIRECT_KEY flag.
_verify_ciphertext_for_encryption_policy Adiantum Adiantum v2
_verify_ciphertext_for_encryption_policy Adiantum Adiantum v2 direct

# success, all done
status=0
exit
