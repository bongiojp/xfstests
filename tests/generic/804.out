QA output created by 804

Verifying ciphertext with parameters:
	contents_encryption_mode: Adiantum
	filenames_encryption_mode: Adiantum
	options: v2

Verifying ciphertext with parameters:
	contents_encryption_mode: Adiantum
	filenames_encryption_mode: Adiantum
	options: v2 direct
